source/Sashimi.Tests.Shared/Server/ActionHandlerTestBuilder.cs

```diff
@@ -25,23 +40,69 @@ public static ActionHandlerTestBuilder<TCalamari> Create<TCalamari>(Type actionH
             return new ActionHandlerTestBuilder<TCalamari>(actionHandlerType);
         }
 
-        public static TestActionHandlerContext<TCalamariProgram> WithPackage<TCalamariProgram>(this TestActionHandlerContext<TCalamariProgram> context, string path)
-            where TCalamariProgram : CalamariFlavourProgram
+        public static TestActionHandlerContext<TCalamariProgram> WithFilesToCopy<TCalamariProgram>(this TestActionHandlerContext<TCalamariProgram> context, string path)
         {
-            context.Variables.Add(KnownVariables.OriginalPackageDirectoryPath, Path.GetDirectoryName(path));
+            if (File.Exists(path))
+            {
+                context.Variables.Add(KnownVariables.OriginalPackageDirectoryPath, Path.GetDirectoryName(path));
+            }
+            else
+            {
+                context.Variables.Add(KnownVariables.OriginalPackageDirectoryPath, path);
+            }
+
             context.Variables.Add(KnownVariables.Action.Packages.PackageId, path);
             context.Variables.Add(KnownVariables.Action.Packages.FeedId, "FeedId");
 
             return context;
         }
+
+        public static TestActionHandlerContext<TCalamariProgram> WithPackage<TCalamariProgram>(this TestActionHandlerContext<TCalamariProgram> context, string packagePath)
+        {
+            context.Variables.Add(KnownVariables.OriginalPackageDirectoryPath, Path.GetDirectoryName(packagePath));
```

Comment:
Instead of doing all this funky work here, why don't we just make the test fixture pass in a `Package` object which has:

- Id
- Version
- Path

Reply:
done